libheif/box.h

```diff
@@ -412,6 +412,23 @@ class Box_ftyp : public Box
 };
 
 
+class Box_free : public Box
```

Comment:
Would it be possible to reuse the existing `Box_other` as the placeholder box instead of introducing a new `Box_free` class?

Reply:
That was my first attempt. 

It didn't work out because some of the properties are marked essential (e.g. imir and irot, plus the codec might end up getting omitted if we support the JPEG case). We have a check for any `Box_other` instances that are marked as essential, since its something we should handle and can't. So a `Box_free` (which is a no-op) seemed better.

The other way to do this would be to use `Box_other` but to check each association against the properties first, and only make it if we need it.

Or we could do both the `free` (either as an explicit `Box_free` box, or as a `Box_other` with `free` parameter), and the check, which would exactly align with the CDAM description.